src/pybind/mgr/k8sevents/module.py

```diff
@@ -409,17 +413,34 @@ def event_body(self):
 
         event_source = client.V1EventSource(component="ceph-mgr", 
                                             host=self.host)
-        return  client.V1Event(
-                    involved_object=obj_ref, 
-                    metadata=obj_meta, 
-                    message=self.message, 
-                    count=self.count, 
-                    type=self.event_type,
-                    reason=self.event_reason,
-                    source=event_source, 
-                    first_timestamp=self.first_timestamp,
-                    last_timestamp=self.last_timestamp
-                )
+
+        # client.V1Event is not available on kubernetes>19.x. Use CoreV1Event instead
+        # since centos8 only provides kubernetes==11.x, we'll need to support both
+        # implementations
+        try:
+            return  client.CoreV1Event(
+                        involved_object=obj_ref,
+                        metadata=obj_meta,
+                        message=self.message,
+                        count=self.count,
+                        type=self.event_type,
+                        reason=self.event_reason,
+                        source=event_source,
+                        first_timestamp=self.first_timestamp,
+                        last_timestamp=self.last_timestamp
+                    )
+        except AttributeError:
+            return client.V1Event(
+                        involved_object=obj_ref,
+                        metadata=obj_meta,
+                        message=self.message,
+                        count=self.count,
+                        type=self.event_type,
+                        reason=self.event_reason,
+                        source=event_source,
+                        first_timestamp=self.first_timestamp,
+                        last_timestamp=self.last_timestamp
+                    )
```

Comment:
right, but here, you're still referring to CoreV1Event

Reply:
it seems as though this block isn't needed.

Reply:
updated, thanks